### ISSUE TITLE
[bug] conan 1.40.2: TypeError: apply() got multiple values for argument 'strip'

### ISSUE THREAD

**Issue by sboehmann** [conan-io/conan#9650]
With conan 1.40.2 on macOS and Linux (Windows not tested) I get this exception:

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/conans/errors.py", line 34, in conanfile_exception_formatter
    yield
  File "/usr/local/lib/python3.9/site-packages/conans/client/source.py", line 149, in _run_source
    conanfile.source()
  File "/Users/stefan/.conan/data/common/1.0.0/mevislab/stable/export/common/commonrecipe.py", line 66, in source
    self.default_source()
  File "/Users/stefan/.conan/data/common/1.0.0/mevislab/stable/export/common/commonrecipe.py", line 171, in default_source
    self.apply_patches(rename_to, strip, fuzz)
  File "/Users/stefan/.conan/data/common/1.0.0/mevislab/stable/export/common/mixin/patch.py", line 40, in apply_patches
    patch(self, **it)
  File "/usr/local/lib/python3.9/site-packages/conan/tools/files/patches.py", line 61, in patch
    if not patchset.apply(root, strip=strip, fuzz=fuzz):
TypeError: apply() got multiple values for argument 'strip'
```

I can fix the error if I replace the line`if not patchset.apply(root, strip=strip, fuzz=fuzz):` 
with `if not patchset.apply(root=root, strip=strip, fuzz=fuzz):`.

**Comment by memsharded** [comment]
Hi @sboehmann 

I cannot fully guess what is happening, it would be great if you could provide what you are doing inside your files ``commonrecipe.py`` and ``patch.py`` and how you are calling that code:

```
  File "/Users/stefan/.conan/data/common/1.0.0/mevislab/stable/export/common/commonrecipe.py", line 66, in source
    self.default_source()
  File "/Users/stefan/.conan/data/common/1.0.0/mevislab/stable/export/common/commonrecipe.py", line 171, in default_source
    self.apply_patches(rename_to, strip, fuzz)
  File "/Users/stefan/.conan/data/common/1.0.0/mevislab/stable/export/common/mixin/patch.py", line 40, in apply_patches
    patch(self, **it)
```

**Comment by sboehmann** [comment]
Sure. CommonRecipe is our common base class for all recipes _(via conan's python_requires)_. It contains a lot of stuff, but the method `default_source()` looks like this:
```
class CommonRecipe(mixin.Patch):
    def default_source(self, rename_from=None, rename_to=None, apply_patches=True, strip=1, fuzz=None):
        rename_to = 'sources' if not rename_to else rename_to

        // ... download and validate checksum,
        // ... uncompress sources and rename the root directory to rename_to
        // ...

        if apply_patches:
            self.apply_patches(rename_to, strip, fuzz) // LINE 171
```

`mixin/patch.py` 

```
from typing import Optional
from conans.errors import ConanException
from conan.tools.files import patch
from conans import ConanFile
from common import utils

class Patch(object):
    def apply_patches(self:ConanFile, base_path:str='sources', strip:int=1, fuzz:Optional[bool]=None):
        patches = self.conan_data.get('patches')
        if not patches:
            return

        entries = patches.get(self.version)
        if not entries:
            return

        for it in entries:
            for key in it:
                it[key] = utils.substitute(self, it[key]) // performs some variable substitution

            if base_path and not 'base_path' in it:
                it['base_path'] = base_path

            if strip and not 'strip' in it:
                it['strip'] = strip

            if fuzz and not 'fuzz' in it:
                it['fuzz'] = fuzz

            patch(self, **it).  // LINE 40
```

Note: We don't use/implement `def layout(self)` anywhere.

**Comment by memsharded** [comment]
Thanks! I managed to reproduce, and fixing it in https://github.com/conan-io/conan/pull/9653, for next 1.41.

**Comment by lasote** [comment]
The fix has been merged and will be available at 1.41. 
Thanks for reporting!
